changedetectionio/tests/test_xpath_selector_unit.py

```diff
@@ -201,3 +201,56 @@ def test_trips(html_content, xpath, answer):
     html_content = html_tools.xpath_filter(xpath, html_content, append_pretty_line_formatting=True)
     assert type(html_content) == str
     assert answer in html_content
+
+DOM_violation_two_html_root_element = """<!DOCTYPE html>
+<html>
+  <body>
+    <h1>Hello world</h1>
+    <p>First paragraph.</p>
+  </body>
+</html>
+<html>
+  <body>
+    <h1>Hello world</h1>
+    <p>Browsers parse this part by fixing it but lxml doesn't and returns two root element node</p>
+    <p>Therefore, if the path is /html/body/p[1], lxml(libxml2) returns two element nodes not one.</p>
+  </body>
+</html>"""
+@pytest.mark.parametrize("html_content", [DOM_violation_two_html_root_element])
+@pytest.mark.parametrize("xpath, answer", [
+    ("/html/body/p[1]", "First paragraph."),
+    ("/html/body/p[1]", "Browsers parse this part by fixing it but lxml doesn't and returns two root element node"),
+    ("//html/body/p[1]", "First paragraph."),
+    ("//html/body/p[1]", "Browsers parse this part by fixing it but lxml doesn't and returns two root element node"),
+    ("//body/p[1]", "First paragraph."),
+    ("//body/p[1]", "Browsers parse this part by fixing it but lxml doesn't and returns two root element node"),
+    ("/html[2]/body/p[1]", "Browsers parse this part by fixing it but lxml doesn't and returns two root element node"),
+    ("//html[2]/body/p[1]", "Browsers parse this part by fixing it but lxml doesn't and returns two root element node"),
+                          ])
+def test_broken_DOM_01(html_content, xpath, answer):
+    # In normal situation, DOM's root element node is only one. So when DOM violation happens, Exception occurs.
+    with pytest.raises(Exception):
```

Comment:
I intentionally add this test to reproduce the problem. 
And, in the future, libxml2 maybe implement "html5". As I posted the issue, this problem will be gone, and this test will fail. The day, please remove these tests.

Reply:
I reread some conversations related to new drafts. When html5 parser for lxml(libxml2) comes out, please don't update it unless the namespace handling is defined.